SmartDeviceLink/public/SDLChoiceSet.m

```diff
@@ -144,6 +123,47 @@ - (void)setHelpList:(nullable NSArray<SDLVRHelpItem *> *)helpList {
     }
 }
 
+#pragma mark - Helpers
+
+/**
+ Check for duplicate choices and voiceCommands
+
+ @param choices The choices you will be adding
+ @return Boolean that indicates whether choices are unique or not
+ */
+-(BOOL)sdl_menuChoiceCellsAreUnique:(NSArray<SDLChoiceCell *> *)choices {
+    NSMutableSet<SDLChoiceCell *> *identicalCellsCheckSet = [NSMutableSet setWithCapacity:choices.count];
+    NSMutableSet<NSString *> *identicalVoiceCommandsCheckSet = [NSMutableSet set];
+    NSUInteger allVoiceCommandsCount = 0;
+    NSUInteger choiceCellWithVoiceCommandCount = 0;
+    for (SDLChoiceCell *cell in choices) {
+        [identicalCellsCheckSet addObject:cell];
+
+        if (cell.voiceCommands == nil) { continue; }
+        [identicalVoiceCommandsCheckSet addObjectsFromArray:cell.voiceCommands];
+        choiceCellWithVoiceCommandCount += 1;
+        allVoiceCommandsCount += cell.voiceCommands.count;
+    }
+
+    if (identicalCellsCheckSet.count < choices.count) {
+        SDLLogE(@"Attempted to create a choice set with duplicate cell text. Cell text must be unique. The choice set will not be set.");
+        return NO;
+    }
+
+    // All or none of the choices must have VR commands
+    if ((choiceCellWithVoiceCommandCount > 0 && choiceCellWithVoiceCommandCount < choices.count)) {
+        SDLLogE(@"If using voice recognition commands, all of the choice set cells must have unique VR commands. There are %lu cells with unique voice commands and %lu total cells. The choice set will not be set.", (unsigned long)choiceCellWithVoiceCommandCount, (unsigned long)choices.count);
+        return NO;
+    }
```

Comment:
```suggestion
```

This is unnecessary because `SDLPreloadChoicesOperation` L156 will add VR to cells if they need it.